[MTIA][RNG] Implement Python RNG API layer

diviramon opened the pull request:
Summary:
The implementation adds 9 RNG functions (get/set_rng_state, manual_seed, seed, initial_seed, and their _all variants) that integrate with the existing C++ generator infrastructure. The lazy initialization pattern ensures RNG operations queue correctly before device initialization and execute during _lazy_init(). We follow the existing CUDA pattern. In addition, we integrated with the general torch.manual_seed and torch.seed functions following the exisiting custom accelerator pattern.

Added comprehensive test coverage for all API calls as well as for the torch integration.

Test Plan:
buck2 test //mtia/host_runtime/torch_mtia/tests:test_torch_mtia_api

https://www.internalfb.com/intern/testinfra/testrun/2533275122809554

buck2 test fbcode//mtia/host_runtime/torch_mtia/tests:test_mtia_rng_generators

https://www.internalfb.com/intern/testinfra/testrun/9288674362815734

Reviewed By: StellarrZ

Differential Revision: D88014851

albanD reviewed (approved):
I assume this is mostly from the cuda version so sounds fair.
Lint needs fixing but then you can merge.

diviramon committed:
[MTIA][RNG] Implement Python RNG API layer (#169982)

Summary:

The implementation adds 9 RNG functions (get/set_rng_state, manual_seed, seed, initial_seed, and their _all variants) that integrate with the existing C++ generator infrastructure. The lazy initialization pattern ensures RNG operations queue correctly before device initialization and execute during _lazy_init(). We follow the existing CUDA pattern. In addition, we integrated with the general torch.manual_seed and torch.seed functions following the exisiting custom accelerator pattern.

Added comprehensive test coverage for all API calls as well as for the torch integration.

Test Plan:
buck2 test //mtia/host_runtime/torch_mtia/tests:test_torch_mtia_api

https://www.internalfb.com/intern/testinfra/testrun/2533275122809554

buck2 test fbcode//mtia/host_runtime/torch_mtia/tests:test_mtia_rng_generators

https://www.internalfb.com/intern/testinfra/testrun/9288674362815734

Reviewed By: StellarrZ

Differential Revision: D88014851
